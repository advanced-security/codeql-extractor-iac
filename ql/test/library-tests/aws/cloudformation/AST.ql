private import iac

query predicate cloudformation(CloudFormation::Document n) { any() }

query predicate resources(CloudFormation::Resource n) { any() }

query predicate resourceProperties(CloudFormation::ResourceProperties n) { any() }

query predicate s3(CloudFormation::S3Bucket n) { any() }

query predicate s3Policy(CloudFormation::S3Bucket n, CloudFormation::S3BucketPolicy p) {
  p = n.getBucketPolicy()
}

query predicate lambda(CloudFormation::LambdaFunction n) { any() }

query predicate ec2SecurityGroup(CloudFormation::EC2SecurityGroup n) { any() }
query predicate ec2SecurityGroupIngress(CloudFormation::EC2SecurityGroupIngress n) { any() }
query predicate ec2SecurityGroupEgress(CloudFormation::EC2SecurityGroupEgress n) { any() }
query predicate iamRole(CloudFormation::IAMRole n) { any() }
query predicate iamStatement(CloudFormation::IAMStatement p){ any() }

query predicate ecsService(CloudFormation::ECSService p){ any() }
query predicate ecsCluster(CloudFormation::ECSCluster p){ any() }
query predicate ecsTaskSet(CloudFormation::ECSTaskSet p){ any() }
query predicate taskDefinition(CloudFormation::TaskDefinition p){ any() }
query predicate containerDefinition(CloudFormation::ContainerDefinition p){ any() }
