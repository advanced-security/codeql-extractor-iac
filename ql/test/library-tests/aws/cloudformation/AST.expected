cloudformation
| IAMRole.yml:1:1:18:74 | CloudFormation Document |
| IngressEgressResources.yml:1:1:19:21 | CloudFormation Document |
| ecs.yml:1:1:92:40 | CloudFormation Document |
| lambda.yml:1:1:21:18 | CloudFormation Document |
| s3.json:1:1:77:1 | CloudFormation Document |
| s3.yml:1:1:42:59 | CloudFormation Document |
| securitygroup.yml:1:1:16:24 | CloudFormation Document |
resources
| IAMRole.yml:5:5:18:74 | CloudFormation IAM Role |
| IngressEgressResources.yml:7:5:13:2 | CloudFormation EC2 Security Group Ingress |
| IngressEgressResources.yml:14:5:19:21 | CloudFormation EC2 Security Group Egress |
| ecs.yml:6:5:38:2 | CloudFormation ECS Task Set |
| ecs.yml:39:5:66:2 | CloudFormation ECS Task Definition |
| ecs.yml:67:5:92:40 | CloudFormation Resource |
| lambda.yml:5:5:21:18 | CloudFormation Lambda Function |
| s3.json:4:21:15:9 | CloudFormation S3 Bucket |
| s3.json:16:25:47:9 | CloudFormation S3 Bucket Policy |
| s3.yml:4:5:12:2 | CloudFormation S3 Bucket |
| s3.yml:13:5:28:28 | CloudFormation S3 Bucket Policy |
| securitygroup.yml:7:5:16:24 | CloudFormation EC2 Security Group |
resourceProperties
| IAMRole.yml:7:7:18:74 | CloudFormation Resource Properties |
| IngressEgressResources.yml:9:7:13:2 | CloudFormation Resource Properties |
| IngressEgressResources.yml:16:7:19:21 | CloudFormation Resource Properties |
| ecs.yml:8:7:38:2 | CloudFormation Resource Properties |
| ecs.yml:41:7:66:2 | CloudFormation Resource Properties |
| ecs.yml:70:7:92:40 | CloudFormation Resource Properties |
| lambda.yml:7:7:21:18 | CloudFormation Resource Properties |
| s3.json:6:27:12:13 | CloudFormation Resource Properties |
| s3.json:18:27:46:13 | CloudFormation Resource Properties |
| s3.yml:6:7:10:4 | CloudFormation Resource Properties |
| s3.yml:15:7:28:28 | CloudFormation Resource Properties |
| securitygroup.yml:9:7:16:24 | CloudFormation Resource Properties |
s3
| s3.json:4:21:15:9 | CloudFormation S3 Bucket |
| s3.yml:4:5:12:2 | CloudFormation S3 Bucket |
s3Policy
| s3.yml:4:5:12:2 | CloudFormation S3 Bucket | s3.yml:13:5:28:28 | CloudFormation S3 Bucket Policy |
lambda
| lambda.yml:5:5:21:18 | CloudFormation Lambda Function |
ec2SecurityGroup
| securitygroup.yml:7:5:16:24 | CloudFormation EC2 Security Group |
ec2SecurityGroupIngress
| IngressEgressResources.yml:7:5:13:2 | CloudFormation EC2 Security Group Ingress |
ec2SecurityGroupEgress
| IngressEgressResources.yml:14:5:19:21 | CloudFormation EC2 Security Group Egress |
iamRole
| IAMRole.yml:5:5:18:74 | CloudFormation IAM Role |
iamStatement
| IAMRole.yml:9:9:16:6 | Version ... -10-17' |
| s3.yml:16:9:28:6 | Id: MyPolicy |
ecsService
| ecs.yml:67:5:92:40 | CloudFormation Resource |
ecsCluster
ecsTaskSet
| ecs.yml:6:5:38:2 | CloudFormation ECS Task Set |
taskDefinition
| ecs.yml:39:5:66:2 | CloudFormation ECS Task Definition |
containerDefinition
| ecs.yml:41:7:66:2 | CloudFormation Resource Properties |
